src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -623,8 +624,10 @@ internal Boolean LoadLocalHeaderExtraFieldAndCompressedBytesIfNeeded()
                 _compressedBytes = new Byte[(_compressedSize / MaxSingleBufferSize) + 1][];
                 for (int i = 0; i < _compressedBytes.Length - 1; i++)
                 {
-                    _compressedBytes[i] = new Byte[MaxSingleBufferSize];
+                    _compressedBytes[i] = ArrayPool<byte>.Shared.Rent(MaxSingleBufferSize);
```

Comment:
That looks to be correct...the original buffer pool had a max size of 2GB and I moved it down arbitrarily; however, if we have a use case for it actually being 2GB then we should move it back up to leverage that scenario. We didn't have data to make a good estimate of what the max buffer size should be, so I feel like moving it up to help this scenario would be beneficial. 

I'm working on getting those numbers now, I put this PR out to get test coverage since the Private PR run seems to have disappeared from Jenkins :(